3.Linear_Modeling/nbconverted/fit_linear_model.py

```diff
@@ -0,0 +1,364 @@
+
+# Define inputs and outputs
+feature_file = pathlib.Path(
+    "../../Extracted_Features_(CSV_files)/interstellar_wave3_sc_norm_cellprofiler.csv.gz"
+)
+feature_df = pd.read_csv(feature_file, engine="pyarrow")
+
+
+# In[4]:
+
+
+print(feature_df.shape)
+feature_df.head()
+
+
+# In[5]:
+
+
+# removing costes features as they behave with great variance across all data
+feature_df = feature_df.drop(feature_df.filter(regex="Costes").columns, axis=1)
+print(feature_df.shape)
+feature_df.head()
+
+
+# In[6]:
+
+
+# replacing '/' in treatment dosage column to avoid errors in file interpolation including such strings
+feature_df = feature_df.replace(to_replace="/", value="_per_", regex=True)
+feature_df.head()
+
+
+# In[7]:
+
+
+# Recycled code from: https://github.com/WayScience/NF1_SchwannCell_data/blob/main/5_analyze_data/notebooks/linear_model/fit_linear_model.ipynb
+cell_count_df = (
+    feature_df.groupby("Metadata_Well")["Metadata_Plate"]
+    .count()
+    .reset_index()
+    .rename(columns={"Metadata_Plate": "Metadata_number_of_singlecells"})
+)
+
+
+# In[8]:
+
+
+feature_df = feature_df.merge(cell_count_df, on="Metadata_Well")
+
+# Drop na and reindex accordingly
+feature_df = (
+    feature_df.assign(
+        Metadata_Treatment_and_Dose=lambda x: feature_df["Metadata_treatment"]
+        + "_"
+        + feature_df["Metadata_dose"]
+    )
+    .dropna()
+    .reset_index()
+)
+
+cp_features = infer_cp_features(feature_df)
+print(f"We are testing {len(cp_features)} CellProfiler features")
+
+feature_df["Metadata_Treatment_and_Dose"].unique()
+
+
+# ##### Here I plot the beta coefficients for each treatment against the number of cells per well. Data points the drift heavily in the Y axis are features that are affected the most by the y-axis treatment while data points that drift more in the x-axis are features that are most affected by the number of cells in a well.
+
+# #### Simple Linear Modeling (cell count beta + 1 beta approach)
+# Here I merged the treatment and dosage and used DMSO 0.1% as the control simply comparing one dosage/treatment at a time and outputting each graph for each treatment for all features. All features and treatments will be exported into 1 file.
+#
+# Linear Model:
+# $y = \beta _{0}x+ \beta _{1}x+ \epsilon$ where;
+# $y$ is each feature
+# $x$ is the inputed variables
+# $\beta _{0}$ is the beta coefficient attributed to cell count,
+# $\beta _{1}$ is the beta coefficient attributed to treatment & dose combined.
+# $\epsilon$ is the residual variance not explained by factors in the model
+
+# In[9]:
+
+
+control = ["DMSO 0.1%_0"]
+# lm_results_df_all = pd.DataFrame(cp_features, columns=["feature"])
+# lm_results_df_all
+lm_results = []
+for i in feature_df["Metadata_Treatment_and_Dose"].unique():
+    treatment = []
+    treatment.append(i)
+
+    dosage_treatments_list = treatment + control
+    print(dosage_treatments_list)
+    df = feature_df.query("Metadata_Treatment_and_Dose == @dosage_treatments_list")
```

Comment:
The following suggestion is to improve readability and make code more pythonic.

```suggestion
control = "DMSO 0.1%_0"

lm_results = []
for i in feature_df["Metadata_Treatment_and_Dose"].unique():
    dosage_treatments_list = [treatment , control]
    print(dosage_treatments_list)
    df = feature_df.query("Metadata_Treatment_and_Dose in @dosage_treatments_list")
```

Note: A couple things changed here:

1. No list in `control` variable (to avoid the need for `treatment.append()`)
2. Removing superfluous comments
3. Update to how dosage_treatments list is constructed
4. update in query using `in` syntax